addons/crm_addons/models/crm_lead.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+class Lead(models.Model):
+    _inherit = 'crm.lead'
+
+    @api.model
+    def form_view_auto_fill(self):
+        return {
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'crm.lead',
+            'target': 'self',
+            'context': {
+                'default_partner_id': self.env.context.get('params', {}).get('partner_id'),
```

Comment:
I still don't understand where this comes from, it really bothers me.

Reply:
OK this comes from the action-loading, which copies `state` (which comes from `bbq.getState`) as part of context.params: https://github.com/odoo/odoo/blob/bb3bc7c9e4992dc07b10e87f87662bdd72f24d88/addons/web/static/src/js/chrome/action_manager_act_window.js#L93

@aab-odoo @ged-odoo is this by design and something the backend can rely on, or not? It's from the big actionmanager refactoring and is a bit odd as it's part of the Act Window override, but it effects a server action, so `action_manager_act_window.js` seems to do quite a bit more processing than strictly just handling the case of an an act_window.

Reply:
@xmo-odoo I don't remember adding this. Typically, at each refactoring, we kind of build the specs by observing the current situation, and we try to reproduce it to not break anything (which almost never works :D). I honestly don't know why we have this in the context, but I found the [commit](c63da2f8a71817301c76889a7d04a8a4ab164447) that introduced it... Maybe @Gorash remembers... ?

This is used for now (at least) at 3 different places in Odoo (mail_thread.py and website.py x2). So I would prefer that no one uses it, so that we can get rid of it, but if there is a need for it (and no alternative), I guess it is fine.

Reply:
@aab-odoo OK wow, that's not new stuff, sorry about the accusation.

Issue's here we want to open a form in creation mode from an external system, *with some default data we provide* (not stuff which is stored / returned by Odoo).

It works by opening the action and getting whatever we need from from the params, but feels a bit iffy. Maybe you know of a better way?

Opening the view is easy, the issue's passing in the defaults, I'm not sure there's any built-in support for that.

Reply:
I would have tried to use `active_id` which seems more appropriate and usual, but I guess you have too, and it doesn't work (?)

Reply:
active_id would be the id of the record to open, here the use case is to open not-a-record (a form in creation mode), with the value we specify as default for one of the field.